database/database_handler.py

```diff
@@ -0,0 +1,88 @@
+import os
+import subprocess
+import json
+
+from sqlalchemy import create_engine, Column, String, Boolean, Date, Integer, Float
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.exc import SQLAlchemyError
+
+from utils.general import LOGGER
+
+LOCAL_RUN = False
+
+
+def get_db_access_token(client_id):
+    # Authenticate using Managed Identity (MSI)
+    try:
+        command = ["az", "login", "--identity", "--username", client_id]
+        subprocess.check_call(command)
+    except subprocess.CalledProcessError as e:
+        print("Error during 'az login --identity':", e)
+        raise e
+
+    # Execute Azure CLI command to get the access token
+    command = ["az", "account", "get-access-token", "--resource-type", "oss-rdbms"]
+    output = subprocess.check_output(command)
+
+    # Parse the output to retrieve the access token
+    access_token = json.loads(output)["accessToken"]
+
+    return access_token
+
+
+def make_connection_string():
+    # Load the JSON file, this is in the root of the Blurring-as-a-Service project
+    with open('database.json') as f:
+        config = json.load(f)
+
+    # Retrieve values from the JSON
+    hostname = config["hostname"]
+    username = config["username"]
+    database_name = config["database_name"]
+    client_id = config["client_id"]
+    password = get_db_access_token(client_id)
+
+    db_url = f"postgresql://{username}:{password}@{hostname}/{database_name}"
+
+    return db_url
+
+
+def create_connection():
+    try:
+        # Create the engine
+        if LOCAL_RUN:
+            db_url = f"postgresql://{os.environ['POSTGRES_USER']}:{os.environ['POSTGRES_PASSWORD']}@{os.environ['POSTGRES_HOST']}/{os.environ['POSTGRES_DB']}"
+        else:
+            db_url = make_connection_string()
+
+        # Create and open a session
+        engine = create_engine(db_url)
+        Session = sessionmaker(bind=engine)
```

Comment:
It would be useful to add `autocommit=False, autoflush=False,` to the sessionmaker to force the user to commit.

Reply:
I like this one. With this we can also do something later with rollback() if an error occurs. I will put this one in Jira